plugins/UserCountry/tests/Integration/APITest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\UserCountry\tests;
+
+use Piwik\Common;
+use Piwik\Plugins\UserCountry\API;
+use Piwik\Plugins\UserCountry\LocationProvider;
+use Piwik\Plugins\UserCountry\LocationProvider\DefaultProvider;
+use Piwik\Tests\Framework\Mock\FakeAccess;
+use Piwik\Tests\Framework\TestCase\IntegrationTestCase;
+
+/**
+ * @group UserCountryAPI
+ * @group APITest
+ * @group Plugins
+ */
+class APITest extends IntegrationTestCase
+{
+    /**
+     * @var API
+     */
+    private $api;
+    
+    public function setUp()
+    {
+        parent::setUp();
+
+        FakeAccess::$superUser = true;
+
+        $this->api = API::getInstance();
+
+        // reset location providers as they might be manipulated by other tests
+        LocationProvider::$providers = null;
+        LocationProvider::getAllProviders();
+    }
+
+    public function test_setLocationProvider()
```

Comment:
I guess we need tests for:
* user is not super user -> exception
* provider is invalid -> exception
* dieIfGeolocationAdminIsDisabled was removed but is needed -> when !UserCountry::isGeoLocationAdminEnabled()  then we should not allow even super user to modify the provider (to maintain the config setting `enable_geolocation_admin = 1` working as expected)